docs/guides/security/firewalld.md

```diff
@@ -0,0 +1,416 @@
+  services: ssh
+  ports:
+  protocols:
+  forward: no
+  masquerade: no
+  forward-ports:
+  source-ports:
+  icmp-blocks:
+  rich rules:
+```
+
+## Establised Related Rules
+
+Although I can find no document that specifically states this, it appears that `firewalld` handles the following `iptables` rule internally by default (if you know that this is incorrect, please correct this!):
+
+`iptables -A INPUT -m state --state ESTABLISHED,RELATED -j ACCEPT`
+
+## Interfaces
+
+By default, `firewalld` will listen on all available interfaces. On a bare-metal server with multiple interfaces facing multiple networks, it will be necessary for you to assign an interface to a zone based on the network it faces.
+
+In our examples, we've not added any interfaces, because we are working with an LXD container for lab testing. We only have one interface to work with. Let's say that your "public" zone needs to be configured to use Ethernet port enp3s0 as this port has the public IP on it, and let's say that your "trusted" and "admin" zones are on the LAN interface, which might be enp3s1.
+
+To assign these zones to the appropriate interface, we would use the following commands:
+
+```
+firewall-cmd --zone=public --change-interface=enp3s0 --permanent
+firewall-cmd --zone=trusted --change-interface=enp3s1 --permanent
+firewall-cmd --zone=admin --change-interface=enp3s1 --permanent
+firewall-cmd --reload
+```
+## Common firewall-cmd Commands
+
+We've used some commands already. Here are a few more common commands and what they do:
+
+| Command                             | Result                                                                                                    |
+|-------------------------------------|-----------------------------------------------------------------------------------------------------------|
+|`firewall-cmd --list-all-zones`      | similar to `firewall-cmd --list-all --zone=[zone]` except it lists *all* of the zones and their contents. |
+|`firewall-cmd --get-default-zone`    | shows the default zone, which is "public" unless it has been changed.                                     |
+|`firewall-cmd --list-services --zone=[zone]`| shows all of the services enabled for the zone.                                                    |
+|`firewall-cmd --list-ports --zone=[zone]`| shows all ports open on the zone.                                                                     |
+|`firewall-cmd --get-active-zones`    | shows the zones that are active on the system, their active interfaces, services, and ports.              |
+|`firewall-cmd --get-services`        | shows all available services possible for use.                                                            |
+|`firewall-cmd --runtime-to-permanent`| if you have entered many rules without the --permanent option, do this before reloading.                  |
+
+There are a great many `firewall-cmd` options not covered here, but this gives you the most used commands.
+
```

Comment:
## Testing connections

System administrators are very fond of the `telnet` command to perform their tests and verify that a port is open and accessible.

Today, the recommended command is `nc` (netcat) which you can use as follows:

```
nc -zv 192.168.1.10 22
```

The `-u` option is use to test udp port by the same way.